task_processing/plugins/mesos/execution_framework.py

```diff
@@ -361,7 +361,10 @@ def statusUpdate(self, driver, update):
             task=task_id
         ))
 
-        self.task_update_queue.put(self.translator(update, task_id))
+        md = self.task_metadata[task_id]
```

Comment:
i wonder why do you think it's an abstraction leak: we're not exposing some internal detail here, we're simply attaching the actual input the user had passed to us. it's like a `source` field, helps understanding what caused the event.